### PR TITLE
Remove moderator password reset, IP ban and mute

### PR DESCRIPTION
**Pull request by DanVanAtta**
Supports migration to MySql DB: #1543
IP ban/mute discussed in: https://github.com/triplea-game/triplea/issues/1488, it's sufficient to have username + mac only and not also IP.

### CODE REVIEW

File: src/main/java/games/strategy/engine/lobby/client/ui/LobbyFrame.java [hunk 1]
```diff
@@ -202,14 +191,9 @@ private void showServerMessage(final LobbyServerProperties props) {
...
-      // too many choices is confusing our mods, just give them 3 choices: name, everything, or cancel.
-      // muteTypes.add("IP Address");
-      // muteTypes.add("Mac Address");
+      muteTypes.add("Mac Address");
       muteTypes.add("Username only");
-      // muteTypes.add("IP, Mac");
-      // muteTypes.add("Name, IP");
-      // muteTypes.add("Name, Mac");
-      muteTypes.add("Name, IP, Mac");
+      muteTypes.add("Name + Mac");
```
Comment (ron-murhammer) [inline src/main/java/games/strategy/engine/lobby/client/ui/LobbyFrame.java]: Probably should make this consistent with the above "Name and Mac"

Comment (ron-murhammer) [inline src/main/java/games/strategy/engine/lobby/client/ui/LobbyFrame.java]: Also "Username only" should match above "Name Only"

File: src/main/java/games/strategy/engine/lobby/server/NullModeratorController.java [hunk 2]
```diff
@@ -97,10 +97,6 @@ public void boot(final INode node) {
     // nothing
   }
 
-  void assertUserIsAdmin() {
```
Comment (ron-murhammer) [inline src/main/java/games/strategy/engine/lobby/server/NullModeratorController.java]: Was this just unused?

Comment (prastle) [inline src/main/java/games/strategy/engine/lobby/server/NullModeratorController.java]: Why ip gone?
It helps with alias tracking

Comment (prastle) [inline src/main/java/games/strategy/engine/lobby/server/NullModeratorController.java]: Password reset rarely used but was useful when they forgot it

Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/lobby/server/NullModeratorController.java]: @ron-murhammer  yes

@prastle  IP ban was discussed in: https://github.com/triplea-game/triplea/issues/1488, it's pretty easy to get around it. On the other hand mac is more difficult. Is there actually a reason to do IP banning and not mac? If we are going to use network identifiers, mac is a better one, it is sticky to a users device and is uniquely identifying. IP can be wrong and sometimes fixed even by resetting your router.

@prastle  re: password reset
First it does not make sense to have that functionality in lobby to reset passwords of players that have logged in. Beyond that, with an update of DB we need to rework it a bit. Short term a sql script can replace the functionality, so hang tight.. Medium/longer term if we share authentication with the NodeBB (new) forum, then password reset can be done via the forums (and also in which case this code would not make any sense anymore anyways and would need to be  redone in any case)

**Comment by prastle** [comment]
i agree easy to get around but generally still same service :) mainly it allowed us to identify strange dudes with mutiple comps that acted similar.  I would say keep it but up to u.

File: src/main/java/games/strategy/engine/lobby/client/ui/LobbyFrame.java (hunk 1 above)
Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/lobby/client/ui/LobbyFrame.java]: Match criteria is a case insenstive contains:
```
        if (selectedMuteType.toLowerCase().contains("name")) {
          controller.muteUsername(clickedOn, null);
        }
```
Consolidated the two lists to one: 8482995

**Comment by prastle** [comment]
i guess to clarify

Ever little bit of info helps when dealing with trolls.  Granted we dont have as many lately.

**Comment by DanVanAtta** [comment]
@prastle  IP is only being removed as a ban/mute options. IP info should still otherwise be visible. Let me know if that does not clear things up. Ban/mute will now only be by name, mac, or both (IP is no longer in that mix)

**Comment by prastle** [comment]
as for passwords it was mainly used to reset peeps that forgot their password and reported in warclub .  ( two sep databases as you know)

**Comment by prastle** [comment]
cleared ty

**Comment by DanVanAtta** [comment]
@ron-murhammer updated: 8482995

**Pull request merged**